example/vectorAdd/src/main.cpp

```diff
@@ -73,21 +73,42 @@ class VectorAddKernel
     }
 };
 
+template<class A, class B>
+#if defined(ALPAKA_ACC_CPU_BT_OMP4_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp4<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp2Blocks<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuFibers<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp2Threads<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuSerial<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuThreads<A,B>;
+#elif defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+using DefaultAcc = alpaka::acc::AccGpuCudaRt<A,B>;
+#else
```

Comment:
Is there any specific rule behind this "default" ordering? Should there be any? Currently `AccCpuSerial` was always used, now it will most probably use `AccCpuOmp4` when not building with GPU support enabled.

Reply:
The ordering is the same as used in cmake, specifically the one you see when you open ccmake. So you need to have all backends in front of the one you want turned off.

Reply:
The order in ccmake seems to be alphabetically as it is not the order they are [mentioned in CMake](https://github.com/ComputationalRadiationPhysics/alpaka/blob/develop/alpakaConfig.cmake#L82-L88). You are also missing `ALPAKA_ACC_GPU_HIP_ENABLE`. Furthermore, the order in CMake still has no meaning and is mainly the order they were added.
We should come up with a good default order. @ComputationalRadiationPhysics/alpaka-maintainers Any wishes?

Reply:
Having a good default order is a very good point.

I would put all GPU-based accelerators first, and then CPU-based. In each group would order by decreasing popularity. So start with GPU CUDA, then in the CPU group start with OpenMP from newest to oldest, then other stuff.